api/src/main/java/org/apache/iceberg/exceptions/EntityNotFoundException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+
+/** Exception raised when an entity is not found. */
+public class EntityNotFoundException extends RESTException implements CleanableFailure {
```

Comment:
Do we need this? I remember @rdblue mentioned a common exception for when a resource can't be found, but it doesn't seem like it's really being used anywhere

Reply:
I am not sure why we added this, I think we can remove it.

Reply:
I'd say let's remove this for now since at the moment it's not being used, I think it's something we can add once there's clarity on how it would get used.

Reply:
Ack